cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ResourceDistributionGoal.java

```diff
@@ -260,6 +260,10 @@ protected void updateGoalState(ClusterModel clusterModel, OptimizationOptions op
     // Log broker Ids over balancing limit.
```

Comment:
Looks like `actionAcceptance` implementation above does not take low utilization threshold into account. It should ensure that if all brokers were already under the low threshold, anything under low threshold is acceptable. Am I missing something?

Reply:
What exactly do you mean by `It should ensure that if all brokers were already under the low threshold, anything under low threshold is acceptable`?

I think it should be that after the proposed action, if the load is not within the range of [`lowerLimit`, `upperLimit`], instead of rejecting the action right away, we check whether the load is under the low threshold. If it is, we accept the action and vice versa. Is it correct?

Reply:
> What exactly do you mean by `It should ensure that if all brokers were already under the low threshold, anything under low threshold is acceptable`?

When `actionAcceptance` is called, suppose the cluster has 3 brokers: broker `B1` has a resource (e.g. `disk` resource) utilization under `LowUtilizationThreshold`, and brokers `B2` and `B3` are above `LowUtilizationThreshold`. This is an example cluster, where not all brokers were already under the low threshold -- i.e. `B2` and `B3` are above it. Hence, `LowUtilizationThreshold` does not apply to _any_ broker in the cluster because of how this threshold is defined in its documentation:

https://github.com/linkedin/cruise-control/blob/5ed774eb5c64f6e57a6a001c1306c520170416c5/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/constants/AnalyzerConfig.java#L141-L147

> I think it should be that after the proposed action, if the load is not within the range of [`lowerLimit`, `upperLimit`], instead of rejecting the action right away, we check whether the load is under the low threshold. If it is, we accept the action and vice versa. Is it correct?

No, because we should apply only the relevant limits. If the cluster has brokers above the lower threshold, then LowUtilizationThreshold is irrelevant. Similarly, if all brokers are under the threshold, we use [0, `LowUtilizationThreshold`] as the acceptable limits as described above.

Reply:
We discussed offline -- see https://github.com/linkedin/cruise-control/pull/1391#discussion_r530608090